hrms/overrides/expense_claim.py

```diff
@@ -0,0 +1,350 @@
+                CustomTask.create_link_doc_task_if_all_parents_closed(
+                    None,
+                    TaskType.APPROVE_EXPENSE_CLAIM,
+                    "Expense Claim",
+                    self.name,
+                    None,
+                    None,
+                    [self.approver_1],
+                )
+            elif self.status == ExpenseClaimConstants.PENDING_APPROVAL_BY_ADMIN:
+                if old_doc.status not in [ExpenseClaimConstants.PENDING_APPROVAL]:
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if not (
+                    frappe.session.user == self.approver_1
+                    or frappe.session.user == "Administrator"
+                ):
+                    frappe.throw(_("Only the added approver can approve the document"))
+                self.status = ExpenseClaimConstants.PENDING_APPROVAL_BY_ADMIN
+                CustomTask.mark_tasks_as_completed(
+                    "Expense Claim", self.name, TaskType.APPROVE_EXPENSE_CLAIM
+                )
+            elif self.status == ExpenseClaimConstants.PENDING_APPROVAL_BY_HR:
+                if old_doc.status not in [ExpenseClaimConstants.PENDING_APPROVAL_BY_ADMIN]:
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if not user_has_role(frappe.session.user, RoleConstants.OFFICE_ADMIN_ROLE):
+                    frappe.throw(_("Only Admin approve the document"))
+            elif self.status == ExpenseClaimConstants.PENDING_PAYMENT:
+                if old_doc.status not in [ExpenseClaimConstants.PENDING_APPROVAL_BY_HR]:
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if not user_has_role(frappe.session.user, RoleConstants.HR_L2_EXPENSE_ROLE):
+                    frappe.throw(
+                        _("Only user having HR L2 Expense Approver role can approve the document")
+                    )
+            elif self.status == ExpenseClaimConstants.PAID:
+                if old_doc.status not in [ExpenseClaimConstants.PENDING_PAYMENT]:
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if old_doc.status in [ExpenseClaimConstants.PENDING_PAYMENT]:
+                    if not user_has_role(frappe.session.user, RoleConstants.FINANCE_ROLE):
+                        frappe.throw(
+                            _(
+                                f"Invalid permission, only user with role {RoleConstants.FINANCE_ROLE} is allowed"
+                            )
+                        )
+            elif self.status == ExpenseClaimConstants.CANCELLED:
+                if old_doc.status not in [
+                    ExpenseClaimConstants.DRAFT,
+                    ExpenseClaimConstants.PENDING_APPROVAL,
+                    ExpenseClaimConstants.PENDING_PAYMENT,
+                    ExpenseClaimConstants.PENDING_APPROVAL_BY_HR,
```

Comment:
add admin approve state

Reply:
Done

Reply:
u did not add role check for that status